src/Traits/AbstractCommandTrait.php

```diff
@@ -86,7 +88,62 @@ protected function getRemoteSettings($host, $user, $keyFile, $app, $timestamp =
                 $defaults['server-' . $key] = $server;
             }
         }
-        return $this->tokenReplace($this->getConfig()->get('remote'), $replacements) + $defaults;
+        $settings = $this->tokenReplace($this->getConfig()->get('remote'), $replacements) + $defaults;
+        return $this->processEnvironmentOverrides($settings);
+    }
+
+    /**
+     * Process environment-specific overrides.
+     *
+     * @param array $settings
+     * @return array
+     *
+     * @see self::getRemoteSettings
+     */
+    protected function processEnvironmentOverrides($settings)
+    {
+        $defaults = [
+            'environment_env_var' => 'HOSTNAME',
+            'environment_matcher' => '\\DigipolisGent\\Robo\\Helpers\\Util\\EnvironmentMatcher::regexMatch',
+        ];
+        $settings += $defaults;
+        if (!isset($settings['environment_overrides']) || !$settings['environment_overrides']) {
+            return $settings;
+        }
+
+        // Get the first server in the list.
```

Comment:
Consider ExtractMethod refactoring into `$server = $this->getFirstServer($settings);` which would replace the redundant comment with self-documenting code.